ynput/ayon-core
# Tray: Singleton tray

## Changelog Description
Tray is now singleton process. Only one tray can be running for AYON url and settings variant (`production`, `staging` or dev variants).

## Additional info
Moved webserver module to tray tool and make it as base functionality of the tool. The webserver is also used to determine if tray is running or not. Tray is storing metadata file containing webserver url, pid and information if initialization finished which is used to determine if is already running.

Added some api functions to be able receive some of these information. The most important are `is_tray_running` and `get_tray_server_url`. The `get_tray_server_url` should be used instead of `os.environ["AYON_WEBSERVER_URL"]`.

## Notes
More features will come in future PRs, based on what we'll need.

## Testing notes:
1. Create ayon-core package and upload to AYON server.
2. Use it in a bundle.
3. Start tray.
4. Try to start tray again, it should not start second one.
